src/app/frontend/filter_dataframe.py

```diff
@@ -60,8 +60,10 @@ def filter_dataframe(df: pd.DataFrame) -> pd.DataFrame:
 
     Returns
     -------
-    pd.DataFrame
-        The filtered DataFrame based on the user-selected criteria.
+    tuple[pd.DataFrame, dict]
+        A tuple containing two elements:
+        - The filtered DataFrame based on the user-selected criteria.
+        - A dictionary containing the applied filters.
```

Comment:
Usually for a tuple, the docstring should be broken up as such:
```suggestion
    pd.DataFrame
        The filtered DataFrame based on the user-selected criteria.
    dict
        A dictionary containing the applied filters.
```

Reply:
Thank you for catching this!